site/docs/guides/evaling-with-harmbench.md

```diff
@@ -0,0 +1,117 @@
+
+When analyzing results across multiple test runs, track how changes to your system affect different categories of attacks. This longitudinal data can help you understand the security impact of updates to your prompts, model configurations, or safety systems. Remember that security improvements in one area might sometimes create unexpected vulnerabilities in another, making comprehensive testing crucial.
+
+## Testing different targets
+
+Promptfoo has built-in support for a wide variety of models such as those from OpenAI, Anthropic, Hugging Face, Deepseek, Ollama and more.
+
+### Ollama Models
+
+First, start your Ollama server and pull the model you want to test:
+
+```bash
+ollama pull llama3.1:8b
+```
+
+Then configure Promptfoo to use it:
+
+```yaml
+targets:
+  - ollama:llama3.1:8b
+```
+
+### Your application
+
+To target an application instead of a model, use the [HTTP Provider](/docs/providers/http/), [Javascript Provider](/docs/providers/custom-api/), or [Python Provider](/docs/providers/python/).
+
+For example, if you have a local API endpoint that you want to test, you can use the following configuration:
+
+```yaml
+targets:
+  - id: https
+    config:
+      url: 'https://example.com/generate'
+      method: 'POST'
+      headers:
+        'Content-Type': 'application/json'
+      body:
+        myPrompt: '{{prompt}}'
+```
+
+## Conclusion and Next Steps
+
+While HarmBench provides valuable insights through its static dataset, it's most effective when combined with other red teaming approaches. promptfoo's plugin architecture allows you to chain multiple evaluation types together, combining HarmBench with plugins that generate dynamic test cases. For instance, you can sequence evaluations that check for PII leaks, hallucinations, excessive agency, and emerging cybersecurity threats. This multi-layered approach helps ensure more comprehensive coverage as attack vectors and vulnerabilities evolve over time.
+
+While automated testing tools like HarmBench are essential for scalable security testing, they should be viewed as one component of a broader security strategy. Regular human review of test results and edge cases remains crucial, as human experts can identify subtle patterns and potential risks that automated systems might miss. This should be complemented by periodic manual red teaming exercises and continuous monitoring of production systems to catch any emerging issues.
+
+Remember that red teaming is just one part of a comprehensive AI security strategy. Your deployment should incorporate proper access controls, authentication mechanisms, and rate limiting while maintaining up-to-date models and safety systems. Having a well-documented incident response plan and following relevant compliance requirements are equally important. Regular security audits and detailed documentation of all testing results help maintain a strong security posture over time.
```

Comment:
this is good advice but not very relevant to harmbench

Reply:
Removed